### PR TITLE
JAMES-2813 extract contract test for task manager

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/task/src/test/java/org/apache/james/task/TaskManagerContract.java [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+        softly.assertThat(entryWithId(list, successfulId))
+            .isEqualTo(TaskManager.Status.COMPLETED);
+        softly.assertThat(entryWithId(list, inProgressId))
+            .isEqualTo(TaskManager.Status.IN_PROGRESS);
+    }
+
+    default TaskManager.Status entryWithId(List<TaskExecutionDetails> list, TaskId taskId) {
+        return list.stream()
+            .filter(e -> e.getTaskId().equals(taskId))
+            .findFirst().get()
```
Comment (Arsnael) [inline server/task/src/test/java/org/apache/james/task/TaskManagerContract.java]: `.get()` should be on next line

File: server/task/src/test/java/org/apache/james/task/TaskManagerContract.java [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+        latch1.countDown();
+        latch2.await();
+
+        assertThat(taskManager.list(TaskManager.Status.WAITING))
+            .extracting(TaskExecutionDetails::getTaskId)
+            .containsOnly(waitingId);
+    }
+
+    @Test
+    default void listShouldAllowToSeeInProgressTasks(CountDownLatch waitingForResultLatch) throws Exception {
```
Comment (Arsnael) [inline server/task/src/test/java/org/apache/james/task/TaskManagerContract.java]: Seems to be a test on successful tasks

File: server/task/src/test/java/org/apache/james/task/TaskManagerContract.java [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+        latch1.countDown();
+        latch2.await();
+
+        assertThat(taskManager.list(TaskManager.Status.FAILED))
+            .extracting(TaskExecutionDetails::getTaskId)
+            .containsOnly(failedId);
+    }
+
+    @Test
+    default void listShouldAllowToSeeSuccessfulTasks(CountDownLatch waitingForResultLatch) throws Exception {
```
Comment (Arsnael) [inline server/task/src/test/java/org/apache/james/task/TaskManagerContract.java]: seems to be a test on in progress tasks

File: server/task/src/test/java/org/apache/james/task/TaskManagerContract.java [hunk 4]
```diff
@@ -334,7 +335,7 @@ default void listShouldAllowToSeeFailedTasks(CountDownLatch waitingForResultLatc
     }
 
     @Test
-    default void listShouldAllowToSeeSuccessfulTasks(CountDownLatch waitingForResultLatch) throws Exception {
+    default void listShouldAllowToSeeInProgressfulTasks(CountDownLatch waitingForResultLatch) throws Exception {
```
Comment (remk) [inline server/task/src/test/java/org/apache/james/task/TaskManagerContract.java]: s/InProgressful/InProgress

**Comment by mbaechler** [comment]
test this please

**Comment by mbaechler** [comment]
test this please

**Pull request merged**